pkg/kubectl/resource_filter.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// FilterFunc is a function that knows how to filter a specific resource kind.
+// It receives a generic runtime.Object which must be type-checked by the function.
+// Returns a boolean value true if a resource is filtered, or false otherwise.
+type FilterFunc func(runtime.Object, PrintOptions) bool
+
+// ResourceFilter filters api resources
+type ResourceFilter interface {
+	Filter(obj runtime.Object) (bool, error)
+	AddFilter(reflect.Type, FilterFunc) error
```

Comment:
If you need to have a helper to filter a list of []runtime.Object, do that instead and have it return the number of omitted, an updated []runtime.Object, or an error.

And really this should be using the visitor pattern we already have in place, but we can come back to that later (visitor should be able to handle this transparently today - it's only because we've accumulated a layer of hacks on get.go that it's not as clean today).